lua/astrocommunity/pack/helm/init.lua

```diff
@@ -1,3 +1,37 @@
+local M = {}
+
+function M.setup()
+  vim.filetype.add {
+    extension = {
+      yaml = M.yaml_filetype,
+      yml = M.yaml_filetype,
+      tmpl = M.tmpl_filetype,
+      tpl = M.tpl_filetype,
+    },
+    filename = {
+      ["Chart.yaml"] = "yaml",
+      ["Chart.lock"] = "yaml",
+    },
+  }
+end
+
+function M.is_helm_file(path)
+  local check = vim.fs.find("Chart.yaml", { path = vim.fs.dirname(path), upward = true })
+  return not vim.tbl_isempty(check)
+end
+
+--@private
+--@return string
+function M.yaml_filetype(path, _) return M.is_helm_file(path) and "helm.yaml" or "yaml" end
+
+--@private
+--@return string
+function M.tmpl_filetype(path, _) return M.is_helm_file(path) and "helm.tmpl" or "template" end
+
+--@private
+--@return string
+function M.tpl_filetype(path, _) return M.is_helm_file(path) and "helm.tmpl" or "smarty" end
```

Comment:
These can be moved into the opts, and convert the opts stuff to the function notation.

Reply:
Are there any examples on how to do that? My lua proficiency level is "copy someone else's code around".

Reply:
Ah, wait a second, I think I got it.

Reply:
Did you mean something like this?

```lua
  {
    "AstroNvim/astrocore",
    ---@type AstroCoreOpts
    opts = {
      autocmds = {
        helm_commentstring = {
          {
            event = "FileType",
            pattern = "helm",
            callback = function() vim.opt_local.commentstring = "{{/* %s */}}" end,
          },
        },
      },
      filetypes = {
        extension = {
          gotmpl = "helm",
          yaml = function(path, _) return M.is_helm_file(path) and "helm.yaml" or "yaml" end,
          yml = function(path, _) return M.is_helm_file(path) and "helm.yaml" or "yaml" end,
          tmpl = function(path, _) return M.is_helm_file(path) and "helm.tmpl" or "template" end,
          tpl = function(path, _) return M.is_helm_file(path) and "helm.tmpl" or "smarty" end,
        },
        filename = {
          ["Chart.yaml"] = "yaml",
          ["Chart.lock"] = "yaml",
        },
        pattern = {
          ["helmfile.*%.ya?ml"] = "helm",
        },
      },
    },
  },
```

I still need to keep the `is_helm_file` function somewhere. And the function for yaml and yml extensions is repeated.

Reply:
No. More like:
```lua
opts = function(_, opts)
-- Do opts stuff here
end
```

And define stuff inside the function.

Reply:
I have updated the code. Now opts is a function which defines local helper functions inside. Did I get your idea right?